website/addons/wiki/model.py

```diff
@@ -34,6 +35,47 @@ class AddonWikiNodeSettings(AddonNodeSettingsBase):
 
     complete = True
     has_auth = True
+    is_publicly_editable = fields.BooleanField(default=False, index=True)
+
+    def set_editing(self, permissions, auth=None, node=None, save=False, log=False):
+        """Set the editing permissions for this node.
+
+        :param permissions: A string, either 'public' or 'private'
+        :param auth: All the auth informtion including user, API key
+        :param node: Node to which self belongs,
+         only necessary if logging is desired
+        :param bool save: Whether to save the privacy change
```

Comment:
Minor quibble - I would change this to "If True, save the node settings object".

The reason for this is that side-effects can in fact be saved, even if `save=False`. For instance - if `log=True`, then a `NodeLog` will be created and saved even if the `Node` never is.